lib/string-checker.js

```diff
@@ -84,18 +85,19 @@ StringChecker.prototype = {
     checkString: function(str, filename) {
         filename = filename || 'input';
         str = str.replace(/^#!?[^\n]+\n/gm, '');
+        var esprimaOptions = extend({
```

Comment:
`... extend({}, this._configuration.getEsprimaOptions(), {...` so that user can't overwrite the required options and that the passed in options aren't modified.